AzureMonitorAgent/ama_tst/modules/logcollector.py

```diff
@@ -0,0 +1,243 @@
+
+    # get all AzureMonitorLinuxAgent-* directory names
+    for config_dir in filter((lambda x : x.startswith("Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-")), os.listdir("/var/lib/waagent")):
+        # collect AMA config and status information for all AzureMonitorLinuxAgent-* directories
+        ver = (config_dir.split('-'))[-1]
+        copy_dircontents(os.path.join("/var/lib/waagent",config_dir,"status"), os.path.join(output_dirpath,ver+"-status"))
+        copy_dircontents(os.path.join("/var/lib/waagent",config_dir,"config"), os.path.join(output_dirpath,ver+"-config"))
+
+    return
+
+
+def collect_arc_logs(output_dirpath):
+    # collect GC Extension logs
+    copy_dircontents("/var/lib/GuestConfig/ext_mgr_logs", os.path.join(output_dirpath,"GC_Extension"))
+    # collect AMA Extension logs
+    for config_dir in filter((lambda x : x.startswith("Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-")), os.listdir("/var/lib/GuestConfig/extension_logs")):
+        # collect AMA config and status information for all AzureMonitorLinuxAgent-* directories
+        ver = (config_dir.split('-'))[-1]
+        copy_dircontents(os.path.join("/var/lib/GuestConfig/extension_logs",config_dir), os.path.join(output_dirpath,ver+"-extension_logs"))
+    
+    # collect logs same to both Arc + Azure VM
+    collect_logs(output_dirpath)
+
+    print("Arc logs collected")
+    return
+
+
+def collect_azurevm_logs(output_dirpath):
+    # collect waagent logs
+    os.mkdir(os.path.join(output_dirpath, "waagent"))
+    for waagent_file in filter((lambda x : x.startswith("waagent.log")), os.listdir("/var/log")):
+        copy_file(os.path.join("/var/log",waagent_file), os.path.join(output_dirpath,"waagent"))
+    # collect AMA Extension logs
+    copy_dircontents("/var/log/azure/Microsoft.Azure.Monitor.AzureMonitorLinuxAgent", os.path.join(output_dirpath,"Microsoft.Azure.Monitor.AzureMonitorLinuxAgent"))
+    
+    # collect logs same to both Arc + Azure VM
+    collect_logs(output_dirpath)
+
+    print("Azure VM logs collected")
+    return
+
+
+
+# Outfile function
+    
+def create_outfile(output_dirpath, vm_type, logs_date):
+    with open(os.path.join(output_dirpath,"amalinux.out"), 'w') as outfile:
+        outfile.write("Log Collection Start Time: {0}\n".format(logs_date))
+        outfile.write("--------------------------------------------------------------------------------\n")
+
+        # detected OS + version
+        vm_dist, vm_ver = helpers.find_vm_distro()
+        if (vm_dist and vm_ver):
+            outfile.write("Linux OS detected: {0}\n".format(vm_dist))
+            outfile.write("Linux OS version detected: {0}\n".format(vm_ver))
+        else:
+            outfile.write("Indeterminate OS.\n")
+
+        # detected package manager
+        pkg_manager = helpers.find_package_manager()
+        if (pkg_manager != ""):
+            outfile.write("Package manager detected: {0}\n".format(pkg_manager))
+        else:
```

Comment:
Do you want to have a fall back here in case distro name match in the helper is returning "" as package manager name but it is still available? In case your helper code needs fixing, we will be still able to get customer env info correctly. Like below (sorry it is in shell script: format):
if [ "$pkg_manager" = "" ]; then
        if [ -x "$(command -v dpkg)" ]; then
            pkg_manager="DPKG"
        elif [ -x "$(command -v rpm)" ]; then
            pkg_manager="RPM"
        fi
    fi
if [ "$pkg_manager" = "" ]; then
        echo "Error: Indeterminate package manager"
        echo "Supported Sytems: 'DPKG' & 'RPM'" 
    fi

Reply:
This is definitely valid, I added this logic into the `find_package_manager()` function (in case the function is used in other locations in the future).